pinecone/client.go

```diff
@@ -107,6 +343,38 @@ func (c *Client) IndexWithAdditionalMetadata(host string, namespace string, addi
 	return idx, nil
 }
 
+// ListIndexes retrieves a list of all indexes in a Pinecone [project].
+//
+// Parameters:
+//   - ctx: A context.Context object controls the request's lifetime, allowing for the request
+//   to be canceled or to timeout according to the context's deadline.
+//
+// Returns a slice of pointers to Index objects or an error.
+//
+// Example:
+//  clientParams := pinecone.NewClientParams{
+//	  ApiKey:    "YOUR_API_KEY",
+//	  SourceTag: "your_source_identifier", // optional
+//	}
+//
+//  pc, err := pinecone.NewClient(clientParams)
+//  if err != nil {
+//    log.Fatalf("Failed to create Client: %v", err)
+//  } else {
+//	  fmt.Println("Successfully created a new Client object!")
+//	}
+//
+//  idxs, err := pc.ListIndexes(ctx)
+//	if err != nil {
+//	  log.Fatalf("Failed to list indexes: %v", err)
+//  } else {
+//	  fmt.Println("Your project has the following indexes:")
+//	  for _, idx := range idxs {
+//	    fmt.Printf("- \"%s\"\n", idx.Name)
+//	  }
+//  }
+//
+// [project]: https://docs.pinecone.io/guides/projects/understanding-projects
```

Comment:
nit: Maybe title this "Understanding Projects"? Just match with the actual article.